controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -782,4 +788,188 @@ export class FederatedGraphRepository {
+      })
+      .from(federatedGraphClients)
+      .where(eq(federatedGraphClients.federatedGraphId, graph.federatedGraph.id));
+
+    const clients: ClientDTO[] = [];
+
+    for (const c of fedGraphClients) {
+      const createdUser = await userRepo.byId(c.createdBy);
+      let updatedUser: UserDTO | null = null;
+      if (c.lastUpdatedBy) {
+        updatedUser = await userRepo.byId(c.lastUpdatedBy);
+      }
+
+      clients.push({
+        id: c.id,
+        name: c.name,
+        createdAt: c.createdAt.toISOString(),
+        lastUpdatedAt: c.lastUpdatedAt?.toISOString() || '',
+        createdBy: createdUser?.email || '',
+        lastUpdatedBy: updatedUser ? updatedUser.email : '',
+      });
+    }
+
+    return clients;
+  }
+
+  public async updatePersistedOperations(federatedGraphName: string, userId: string, clientId: string, operations: {hash: string, filePath: string;}[]) {
+    const graph = await this.db.query.targets.findFirst({
+      columns: {},
+      with: {
+        federatedGraph: {
+          columns: { id: true },
+        },
+      },
+      where: and(
+        eq(schema.targets.name, federatedGraphName),
+        eq(schema.targets.organizationId, this.organizationId),
+        eq(schema.targets.type, 'federated'),
+      ),
+    });
+    if (graph === undefined) {
+      throw new Error(`could not find graph ${federatedGraphName}`);
+    }
+    const now = new Date()
+    for (const operation of operations) {
+      await this.db.insert(federatedGraphPersistedOperations).values({
+        federatedGraphId: graph.federatedGraph.id,
+        clientId,
+        hash: operation.hash,
+        filePath: operation.filePath,
+        createdAt: now,
+        updatedAt: now,
+        createdBy: userId,
+        updatedBy: userId,
+      }).onConflictDoUpdate({ target: [federatedGraphPersistedOperations.federatedGraphId, federatedGraphPersistedOperations.hash],  set: { updatedAt: now, updatedBy: userId } });
+    }
+  }
+
+  public async getPersistedOperations(federatedGraphName: string): Promise<PersistedOperationDTO[]> {
+    const graph = await this.db.query.targets.findFirst({
+      columns: {},
+      with: {
+        federatedGraph: {
+          columns: { id: true },
+        },
+      },
+      where: and(
+        eq(schema.targets.name, federatedGraphName),
+        eq(schema.targets.organizationId, this.organizationId),
+        eq(schema.targets.type, 'federated'),
+      ),
+    });
+    if (graph === undefined) {
+      throw new Error(`could not find graph ${federatedGraphName}`);
+    }
+
+    const operationsResult = await this.db
+      .select()
+      .from(federatedGraphPersistedOperations)
+      .where(eq(federatedGraphPersistedOperations.federatedGraphId, graph.federatedGraph.id));
+
+    const operations: PersistedOperationDTO[] = [];
+
+    const userRepo = new UserRepository(this.db);
+    const users = new Map<string, UserDTO>();
+    const getUser = async (userId: string) => {
```

Comment:
Same here. You can do a inner join to the user table.

Reply:
Done